### DISCUSSION TITLE
How to save scroll state in Jetpack Compose

### DISCUSSION THREAD

**Discussion by NielsMasdorp** [patrykandpatrick/vico#317] [Ideas]
Hi, 

Apologies if this is not the right place to ask, but I am fairly new to Jetpack Compose and want to show a chart with the last known scroll state as the device changes orientation etc. I see there is an `ChartScrollState` which I should be able to use but I can't seem to get it to select a given scroll state upon composition. I've tried using a `ScrollListener` and persisting the value in `rememberSaveable {}`/`ViewModel` but it seems to always load to `0f` upon composition. 

Thanks in advance!

**Comment by patrickmichalik** [comment]
Hello! At the moment, `rememberChartScrollState` uses `remember`, not `rememberSaveable`. Admittedly, this isn’t optimal, as configuration changes cause the scroll state to be lost. We’ll be looking into making `rememberChartScrollState` use `rememberSaveable`. As such, one can create and store a `ChartScrollState` instance in a `ViewModel`, in which case everything will be persisted across configuration changes. I’m moving this discussion to the “Ideas” category.

**Comment by patrickmichalik** [comment]
[Vico 2.0.0 Alpha 9](https://github.com/patrykandpatrick/vico/releases/tag/v2.0.0-alpha.9) addresses this. Cheers!

**Comment by Gowsky** [comment]
Hello, [Vico 2.0.0](https://github.com/patrykandpatrick/vico/releases/tag/v2.0.0) is now available.
